### DISCUSSION TITLE
Need a method remove all keys but keep the file size not changed

### DISCUSSION THREAD

**Discussion by calvin2021y** [Tencent/MMKV#1134] [Ideas]
This is not a bug report but a future request.

I need a method to remove all keys, but not trim and truncate file size to DEFAULT_MMAP_SIZE.

any suggestions how to do it myself, or MMKV will consider add this future?

**Comment by calvin2021y** [comment]
I need reuse the db instance without close it. keep the file size to reduce the IO overhead.

**Comment by lingol** [comment]
What do you mean by saying reduce IO overhead? Do you find MMKV's space expansion algorithm not working for you?

**Comment by calvin2021y** [comment]
The file size reduced into 4KB after clear.

I use 2 MMKV instance as log rotate at high speed,  after one full then change to other one. 

In this case every time rotate the file size changed,  I guess keep the file size not change but destroy the data inside it will avoid a lot system call and block time.

**Comment by lingol** [comment]
Why do you need two instances and rotate them? It's quite a strange usage.

**Comment by calvin2021y** [comment]
1) avoid thread lock,  the log always read/write in one thread

2) if I need destroy one, no need search by key range

3) when need backup or send to remote,  just move the task into other thread.

**Comment by lingol** [comment]
I take it that you have set a size limit or key count limit per instance. How large is one file grow to, normally speaking?

**Comment by calvin2021y** [comment]
Yes, I use count limit.  the file size could be 10M to 100M depend the logs item size and MAX_LOG_ENTRY_LIMIT(adjustable).

**Comment by kaitian521** [comment]
Same feature is needed for me.

A faster `clearAll()` is somehow like `clearMemoryCache`,  but do not call `doCleanMemoryCache` in m_file.

In another word, it will not do any system IO calls for better performance.

If some groups of keys are inserted into MMKV together every time, we can first call `clearAllFast`, then put them from start of file: it will not expand, will not trim, just keep it origin size which is expected.
